smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy

```diff
@@ -0,0 +1,869 @@
+                    "url": "https://example.com",
+                    "properties": {
+                        "splitResult": "p1=a; p2=b; p3=c--d; p4=<null>; p5=<null>"
+                    }
+                }
+            }
+        },
+
+        // Limit 4 tests
+        {
+            "documentation": "S3 Express bucket pattern",
+            "params": {
+                "Input": "--x-s3--azid--suffix",
+                "Delimiter": "--",
+                "Limit": "4"
+            },
+            "expect": {
+                "endpoint": {
+                    "url": "https://example.com",
+                    "properties": {
+                        "splitResult": "p1=; p2=x-s3; p3=azid; p4=suffix; p5=<null>"
+                    }
+                }
+            }
+        },
+        {
+            "documentation": "limit 4 stops at 4 parts",
+            "params": {
+                "Input": "a--b--c--d--e",
+                "Delimiter": "--",
+                "Limit": "4"
+            },
+            "expect": {
+                "endpoint": {
+                    "url": "https://example.com",
+                    "properties": {
+                        "splitResult": "p1=a; p2=b; p3=c; p4=d--e; p5=<null>"
+                    }
+                }
+            }
+        },
+
+        // Unicode tests
+        {
+            "documentation": "unicode emoji delimiter",
+            "params": {
+                "Input": "a🌟b🌟c",
+                "Delimiter": "🌟",
+                "Limit": "0"
+            },
+            "expect": {
+                "endpoint": {
+                    "url": "https://example.com",
+                    "properties": {
+                        "splitResult": "p1=a; p2=b; p3=c; p4=<null>; p5=<null>"
+                    }
+                }
+            }
+        },
+
+        // Regex treated literally
+        {
+            "documentation": "regex-like pattern treated literally",
+            "params": {
+                "Input": "a.*b.*c",
+                "Delimiter": ".*",
+                "Limit": "0"
+            },
+            "expect": {
+                "endpoint": {
+                    "url": "https://example.com",
+                    "properties": {
+                        "splitResult": "p1=a; p2=b; p3=c; p4=<null>; p5=<null>"
+                    }
+                }
+            }
+        },
+        {
+            "documentation": "pipe delimiter",
+            "params": {
+                "Input": "a|b|c|d|e",
+                "Delimiter": "|",
+                "Limit": "0"
+            },
+            "expect": {
+                "endpoint": {
+                    "url": "https://example.com",
+                    "properties": {
+                        "splitResult": "p1=a; p2=b; p3=c; p4=d; p5=e"
+                    }
+                }
+            }
+        },
+
+        // Edge cases
+        {
+            "documentation": "single character delimiter",
```

Comment:
not sure I'd consider a single character delimiter as an edge case?

Reply:
Agreed, and I'm removing this test since it's well covered by other stuff